packages/interactivity/src/vdom.ts

```diff
@@ -72,18 +72,13 @@ export function toVdom( root: Node ): Array< ComponentChild > {
 
 		// CDATA_SECTION_NODE (4)
 		if ( nodeType === 4 ) {
-			const next = treeWalker.nextSibling();
-			( node as CDATASection ).replaceWith(
-				new window.Text( ( node as CDATASection ).nodeValue ?? '' )
-			);
-			return [ node.nodeValue, next ];
+			return [ node.nodeValue ];
```

Comment:
Maybe you forgot to add the node to `nodesToReplace`?
```suggestion
			nodesToReplace.add( node );
			return [ node.nodeValue ];
```

Reply:
Good catch, fixed in https://github.com/woocommerce/gutenberg/pull/7/commits/78bd2db89ddf2a30c79ac7eb9cdf29c6fc55527d and added test coverage.